twine/commands/upload.py

```diff
@@ -58,21 +58,37 @@ def upload(upload_settings: settings.Settings, dists: List[str]) -> None:
     uploads = [i for i in dists if not i.endswith(".asc")]
     upload_settings.check_repository_url()
     repository_url = cast(str, upload_settings.repository_config["repository"])
+    packages_to_upload = []
+
+    for filename in uploads:
+        package = package_file.PackageFile.from_filename(
+            filename, upload_settings.comment
+        )
+
+        signed_name = package.signed_basefilename
+
+        if signed_name in signatures:
+            package.add_gpg_signature(signatures[signed_name], signed_name)
+        elif upload_settings.sign:
+            package.sign(upload_settings.sign_with, upload_settings.identity)
+
+        packages_to_upload.append(package)
+
+    file_package_zip = zip(uploads, packages_to_upload)
 
     print(f"Uploading distributions to {repository_url}")
     if upload_settings.verbose:
-        for filename in uploads:
+        for filename, package in file_package_zip:
             file_size = utils.get_file_size(filename)
             print(f"  {filename} ({file_size})")
+            print(f"  {package.signed_filename} (Signature)")
```

Comment:
Not ever package will have a signed filename. What happens here if it doesn't have a signature file? What does this print? Folks generating signatures should also know that a `.asc` file is the signature, what are we hoping to communicate by putting `(Signature)` at the end?

Reply:
That makes sense, thanks for the feedback. Would you suggest printing the filename like the other files(without anything to indicate that its a signature)? I will also make this print conditional on upload_settings.sign